### PR TITLE
Upgrade various build and doc links to https.

### PR DESCRIPTION
**Pull request by gianm**
Where it wasn't possible to upgrade build-time dependencies to https,
I kept http in place but used hardcoded checksums or GPG keys to ensure
that artifacts fetched over http are verified properly.

### CODE REVIEW

File: integration-tests/docker-base/setup.sh [hunk 1]
```diff
@@ -34,14 +34,30 @@ apt-get install -y mysql-server
 apt-get install -y supervisor
 
 # Zookeeper
-wget -q -O - http://www.us.apache.org/dist/zookeeper/zookeeper-3.4.13/zookeeper-3.4.13.tar.gz | tar -xzf - -C /usr/local \
-  && cp /usr/local/zookeeper-3.4.13/conf/zoo_sample.cfg /usr/local/zookeeper-3.4.13/conf/zoo.cfg \
-  && ln -s /usr/local/zookeeper-3.4.13 /usr/local/zookeeper
+wget -q -O /tmp/zookeeper-3.4.14.tar.gz "http://www.us.apache.org/dist/zookeeper/zookeeper-3.4.14/zookeeper-3.4.14.tar.gz"
```
Comment (b-slim) [inline integration-tests/docker-base/setup.sh]: how about using this one https://apache.org/dist/zookeeper/zookeeper-3.4.14/zookeeper-3.4.14.tar.gz

Comment (gianm) [reply integration-tests/docker-base/setup.sh]: Sure, is it better for some reason?

File: examples/quickstart/tutorial/hadoop/docker/setup-zulu-repo.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Script to set up the Azul Zulu JDK yum repository.
+#
+
+# Hardcode GPG key so we don't have to fetch it over http.
```
Comment (xvrl) [inline examples/quickstart/tutorial/hadoop/docker/setup-zulu-repo.sh]: how can you be sure the one you got was legit ;) ?

File: integration-tests/docker-base/setup.sh (hunk 1 above)
Comment (himanshug) [reply integration-tests/docker-base/setup.sh]: probably because that is "https" , we can probably do same for kafka url below.

Comment (gianm) [reply integration-tests/docker-base/setup.sh]: Ah I didn't notice the https. I guess I can remove the shasum checks too. What do you all think?

Comment (himanshug) [reply integration-tests/docker-base/setup.sh]: sure

File: examples/quickstart/tutorial/hadoop/docker/setup-zulu-repo.sh (hunk 2 above)
Comment (himanshug) [reply examples/quickstart/tutorial/hadoop/docker/setup-zulu-repo.sh]: if we are using them only for jre , we can directly download same from https://cdn.azul.com/zulu/bin/zulu8.38.0.13-ca-jre8.0.212-linux_x64.tar.gz and not configure the yum repo.
Ref: https://www.azul.com/downloads/zulu/

Comment (gianm) [reply examples/quickstart/tutorial/hadoop/docker/setup-zulu-repo.sh]: Do you mind if we keep the yum repo? That way, it'll always use the latest jdk.

File: integration-tests/docker-base/setup.sh (hunk 1 above)
Comment (gianm) [reply integration-tests/docker-base/setup.sh]: OK, updated.

File: examples/quickstart/tutorial/hadoop/docker/setup-zulu-repo.sh (hunk 2 above)
Comment (himanshug) [reply examples/quickstart/tutorial/hadoop/docker/setup-zulu-repo.sh]: Ah, now I see why we are using yum there. sure, feel free to keep it in that case.

**Comment by fjy** [comment]
Merging given that TC is broken.

**Pull request merged**